python/test/test_onnx.py

```diff
@@ -1152,6 +1155,57 @@ def test_Relu(self, case_name):
                                       initializer=[weight, bias])
         self.onnx_and_test(graph_def)
 
+    def test_PermuteMove(self, case_name):
+        input_shape = [1, 16, 28, 28]
+        input2_shape = [1, 1, 28, 28]
+        output_shape = [16, 1, 28, 28]
+
+        input = helper.make_tensor_value_info('input', TensorProto.FLOAT, input_shape)
+        input2 = helper.make_tensor_value_info('input2', TensorProto.FLOAT, input2_shape)
+        transpose_node = helper.make_node("Transpose",
+                                         inputs=['input'],
+                                         outputs=['e'],
+                                         perm=[1, 0, 2, 3])
+        output = helper.make_tensor_value_info('output', TensorProto.FLOAT, output_shape)
+        w_data = np.random.rand(*output_shape).astype(np.float32)
+        w_value = helper.make_tensor(
+            name='w',
+            data_type=onnx.TensorProto.FLOAT,
+            dims=w_data.shape,
+            vals=w_data.flatten(),
+        )
+
+        for i in ['Relu', 'Sigmoid']:
+            add_node = helper.make_node(
+                i,  # node name
+                ['e'],  # inputs
+                ['output'],  # outputs
+            )
+            graph_def = helper.make_graph([transpose_node, add_node],
+                                          case_name + '_' + i, [input], [output])
+            self.onnx_and_test(graph_def, check_last=True)
+
+        # add cast
+        add_node = helper.make_node(
+                'Add',  # node name
+                ['e', 'input2'],  # inputs
+                ['output'],  # outputs
+                )
+        graph_def = helper.make_graph([transpose_node, add_node],
+                                      case_name + '_' + 'AddCast', [input, input2], [output])
+        self.onnx_and_test(graph_def, check_last=True)
+
+        # addconst
+        add_node = helper.make_node(
+                'Add',  # node name
+                ['e', 'w'],  # inputs
+                ['output'],  # outputs
+                )
+        graph_def = helper.make_graph([transpose_node, add_node],
```

Comment:
In fact it does not invoke `AddToAddConst` pass correctly.

```
module attributes {module.chip = "ALL", module.name = "AddConst", module.platform = "ONNX", module.state = "TOP_F32", module.weight_file = "addconst_top_f32_all_origin_weight.npz"} {
  func.func @main(%arg0: tensor<1x16x28x28xf32> loc(unknown)) -> tensor<1x16x28x28xf32> {
    %0 = "top.None"() : () -> none loc(#loc)
    %1 = "top.Input"(%arg0) : (tensor<1x16x28x28xf32>) -> tensor<1x16x28x28xf32> loc(#loc1)
    %2 = "top.Weight"() : () -> tensor<1x16x28x28xf32> loc(#loc2)
    %3 = "top.Add"(%1, %2) {do_relu = false, relu_limit = -1.000000e+00 : f64} : (tensor<1x16x28x28xf32>, tensor<1x16x28x28xf32>) -> tensor<1x16x28x28xf32> loc(#loc3)
    return %3 : tensor<1x16x28x28xf32> loc(#loc)
  } loc(#loc)
} loc(#loc)
```

While my impl should be correct, I can't test it. Debugger showed that it failed here: https://github.com/sophgo/tpu-mlir/blob/master/lib/Dialect/Top/Canonicalize/Add.cpp#L41-L59

Reply:
For this part that's because AddConst means input plus a single_value, but you set weight as a tensor with the same shape of output, so it doesn't invoke AddToAddConst. 

Maybe you were misdirected by the test_AddConst(), which is a typo. It should be called 'test_AddWeight()' so I have to apologize for that.

Reply:
For this part that's because AddConst means input plus a single_value, but you set weight as a tensor with the same shape of output, so it doesn't invoke AddToAddConst. 

Maybe you were misdirected by the test_AddConst(), which is a typo and should be corrected as 'test_AddWeight()', so I have to apologize for that.

Reply:
> For this part that's because AddConst means input plus a single_value, but you set weight as a tensor with the same shape of output, so it doesn't invoke AddToAddConst.
> 
> Maybe you were misdirected by the test_AddConst(), which is a typo and should be corrected as 'test_AddWeight()', so I have to apologize for that.

You're welcome. Glad to hear that.